crates/biome_js_analyze/tests/specs/nursery/noGlobalEval/valid.js

```diff
@@ -0,0 +1,121 @@
+/* should not generate diagnostics */
+var foo = "foo";
+Eval(foo);
+
+setTimeout("foo");
+
+setInterval("foo");
+
+window.setTimeout("foo");
+
+window.setInterval("foo");
+
+window.noeval("foo");
+
+global.noeval("foo");
+
+globalThis.noeval("foo");
+
+this.noeval("foo");
+
+function foo() {
+	"use strict";
+	this.eval("foo");
+}
+
+("use strict");
+this.eval("foo");
+
+function foo() {
+	this.eval("foo");
+}
+
+var obj = {
+	foo: function () {
+		this.eval("foo");
+	},
+};
+
+var obj = {};
+obj.foo = function () {
+	this.eval("foo");
+};
+
+() => {
+	this.eval("foo");
+};
+
+function f() {
+	"use strict";
+	() => {
+		this.eval("foo");
+	};
+}
+
+(function f() {
+	"use strict";
+	() => {
+		this.eval("foo");
+	};
+});
+
+class A {
+	foo() {
+		this.eval("foo");
+	}
+}
+
+class A {
+	static foo() {
+		this.eval();
+	}
+}
+
+class A {
+	field = this.eval();
+}
+
+class A {
+	field = () => this.eval();
+}
+
+class A {
+	static {
+		this.eval();
+	}
+}
+
+Array.findLast(
+	function (x) {
+		return this.eval.includes(x);
+	},
+	{ eval: ["foo", "bar"] },
+);
+
+callbacks.findLastIndex(function (cb) {
+	return cb(this.eval);
+}, this);
+
+["1+1"].flatMap(function (str) {
+	return this.eval(str);
+}, new Evaluator());
+
+// TODO Fix to prevent errors for these cases
+// function foo() {
+// 	var eval = "foo";
+// 	window[eval]("foo");
+// }
+
+// function foo() {
+// 	var eval = "foo";
+// 	global[eval]("foo");
+// }
+
+// function foo() {
+// 	var eval = "foo";
+// 	globalThis[eval]("foo");
+// }
+
+// function foo(eval) {
+// 	eval("var a = 0");
+// }
```

Comment:
Following four test cases, eval is bound to local scope variables, and the global eval is not being called. Therefore, the error for noGlobalEval should not be displayed. I understand that if eval is bound to a local scope variable, it should be treated as None with the following process, but it seems not to work as expected.
```rs
model.binding(&reference).is_none().then_some(())
```

Reply:
I think the bug is coming from the semantic model. I will investigate the case.